pallets/settlement/src/lib.rs

```diff
@@ -1308,13 +1563,25 @@ impl<T: Trait> Module<T> {
                 }
                 LegStatus::ExecutionPending => {
                     // This can never return an error since the settlement module
-                    // must've locked these tokens when instruction was authorized
-                    T::Portfolio::unlock_tokens(
-                        &leg_details.from,
-                        &leg_details.asset,
-                        &leg_details.amount,
-                    )
-                    .ok();
+                    // must've locked these tokens when instruction was authorized.
+                    match leg_details {
+                        Leg::NonConfidentialLeg(non_confidential_leg_detail) => {
+                            T::Portfolio::unlock_tokens(
+                                &non_confidential_leg_detail.from,
+                                &non_confidential_leg_detail.asset,
+                                &non_confidential_leg_detail.amount,
+                            )
+                            .ok();
+                        }
+                        Leg::ConfidentialLeg(_) => {
+                            // TODO: should I dispatch error at this point since the function does not have an error type?
+                            // return Err(Error::<T>::ConfidentialModeNotSupportedYet.into())
+                        }
+                        Leg::Undefined => {
+                            // TODO: should I dispatch error at this point since the function does not have an error type?
+                            // return Err(Error::<T>::UndefinedLeg.into())
+                        }
```

Comment:
More error handling todos.

Reply:
CC @vkomenda @satyamakgec ?

Reply:
Yeah, we don't emit error here as it is `unchecked` prefix based function. But the idea here is that we never hit those conditions here which lead to errors.